### PR TITLE
Use names for integration_method

### PR DESCRIPTION
**Pull request by WeinaJi**

### CODE REVIEW

**Comment by WeinaJi** [comment]
This change is in response to https://github.com/BlueBrain/snap/issues/275

**Comment by mgeplf** [comment]
Thanks!

Since we're changing this, what do you think about using names instead of integers? That makes it much easier to understand in the config, and would result in fewer errors, I think.

File: source/sonata_simulation.rst [hunk 1]
```diff
@@ -81,7 +81,7 @@ Parameters defining global simulation settings. As NEURON is the engine used for
...
    spike_threshold                 float      Optional    The spike detection threshold. A spike is detected whenever the voltage in the spike detection location goes over the spike threshold value. Default is -30.0 mV. `NEURON specific details <https://nrn.readthedocs.io/en/latest/python/modelspec/programmatic/network/netcon.html#NetCon.threshold>`_.
-   integration_method              text       Optional    Selects the NEURON/CoreNEURON `integration method <https://nrn.readthedocs.io/en/latest/python/simctrl/programmatic.html#secondorder>`_. This parameter sets the NEURON global variable h.secondorder. The allowed values are ‘0’ (default) for fully implicit backward euler, ‘1’ for Crank-Nicolson and ‘2’ for Crank-Nicolson with fixed ion currents.
+   integration_method              integer    Optional    Selects the NEURON/CoreNEURON `integration method <https://nrn.readthedocs.io/en/latest/python/simctrl/programmatic.html#secondorder>`_. This parameter sets the NEURON global variable h.secondorder. The allowed values are 0 (default) for fully implicit backward euler, 1 for Crank-Nicolson and 2 for Crank-Nicolson with fixed ion currents.
```
Comment (seirios) [inline source/sonata_simulation.rst:84]: Minor typo, it's "Nicholson"

**Review by seirios — COMMENTED** [review]
I think using integers would be familiar to NEURON users, but it's true that strings are harder to mess up.
If names were preferred, I'd suggest: "euler", "crank_nicolson", "crank_nicolson_ion".

File: source/sonata_simulation.rst (hunk 1 above)
Comment (seirios) [reply source/sonata_simulation.rst:84]: I correct myself, "Nicolson" is correct, it's the NEURON docs that have it wrong :p

Comment (seirios) [reply source/sonata_simulation.rst:84]: I prefer names, it's less cryptic and the relation to NEURON's `secondorder` is clear from the documentation.

**Review by seirios — APPROVED** [review]
LGTM

**Review by mgeplf — APPROVED** [review]

**Pull request merged**